lib/ruby/truffle/truffle/bigdecimal.rb

```diff
@@ -6,10 +6,68 @@ class Truffle::BigDecimal < Numeric
   SIGN_NEGATIVE_INFINITE = -3
   SIGN_NEGATIVE_FINITE   = -2
   SIGN_NEGATIVE_ZERO     = -1
-  SIGN_NaN               =  0
-  SIGN_POSITIVE_ZERO     =  1
-  SIGN_POSITIVE_FINITE   =  2
-  SIGN_POSITIVE_INFINITE =  3
+  SIGN_NaN               = 0
+  SIGN_POSITIVE_ZERO     = 1
+  SIGN_POSITIVE_FINITE   = 2
+  SIGN_POSITIVE_INFINITE = 3
+
+  EXCEPTION_INFINITY   = 1
+  EXCEPTION_OVERFLOW   = 1
+  EXCEPTION_NaN        = 2
+  EXCEPTION_UNDERFLOW  = 4
+  EXCEPTION_ZERODIVIDE = 16
+  EXCEPTION_ALL        = 255
+  ROUND_MODE           = 256
+
+  ROUND_UP        = 1
+  ROUND_DOWN      = 2
+  ROUND_HALF_UP   = 3
+  ROUND_HALF_DOWN = 4
+  ROUND_CEILING   = 5
+  ROUND_FLOOR     = 6
+  ROUND_HALF_EVEN = 7
+
+  def self.mode(key, value = nil)
+    raise ArgumentError, 'requires key to be Fixnum' unless key.is_a? Fixnum
+    if key == ROUND_MODE
+      Thread.current[:'BigDecimal.rounding_mode'] ||= 3
```

Comment:
yeah but we found out with @eregon in Linz that CRuby leaks the value under the same exact name :) Would it still make sense that we'll hide it?